messaging/src/main/java/org/axonframework/eventhandling/TrackingToken.java

```diff
@@ -68,4 +68,24 @@ default OptionalLong position() {
         return OptionalLong.empty();
     }
 
+    /**
+     * Indicates whether this token is at the exact same position as the {@code other} token.
+     * This method is similar to {@link #covers(TrackingToken)}, but for some token implementations (like
+     * {@link GapAwareTrackingToken}) it may use a less strict comparison that only checks the index position,
+     * ignoring gap structures.
+     * <p>
+     * This method is particularly useful when comparing tokens from different points in time, such as during
+     * replay detection, where gap structures may naturally differ.
+     * <p>
+     * By default, this method checks bidirectional coverage: {@code this.covers(other) && other.covers(this)},
+     * which ensures both tokens are at the same position.
+     *
+     * @param other The token to compare to this one
+     * @return {@code true} if this token's position equals the other's position, otherwise {@code false}
+     * @see #covers(TrackingToken)
```

Comment:
I don't think we have to take gaps as the direct example here. Given the generic nature of the `TrackingToken`, I'd rather not, actually. Same with use of position in this description, which I think we should steer away from as a token itself already is something like a position or place in time/stream. 

So, my two cents on the JavaDoc for this method:

```suggestion
     * Indicates whether {@code this} token is at the exact same spot in the event stream as the {@code other} token.
     * <p>
     * This method is particularly useful when comparing tokens from different points in time, such as during
     * replay detection, where token implementations may naturally differ.
     * <p>
     * By default, this method checks bidirectional coverage: {@code this.covers(other) && other.covers(this)},
     * which ensures both tokens are at the same position.
     *
     * @param other The token to validate against {@code this} token's
     * @return {@code true} if this token's position equals the other's position, otherwise {@code false}
     * @see #covers(TrackingToken)
```